torchao/quantization/quantize_/common/kernel_preference.py

```diff
@@ -0,0 +1,41 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD 3-Clause license found in the
+# LICENSE file in the root directory of this source tree.
+
+from enum import Enum
+
+import torch
+
+from torchao.utils import TORCH_VERSION_AT_LEAST_2_5
+
+
+# can switch to StrEnum (https://docs.python.org/3/library/enum.html#enum.StrEnum)
+# after python 3.10 is end of life (https://devguide.python.org/versions/)
+class KernelPreference(str, Enum):
+    """Enum for specifying the groups of kernels that's used for quantization, matrix multiplication
+    or other compute ops for quantized tensor
+    """
+
+    """Use the most efficient quantize and mm kernels chosen for user based on hardware and library availabilities and versions etc.
+
+    Examples of how this option affects the selected kernels can be found in tensor subclass implementations under torchao/quantization/quantize_/workflows
```

Comment:
nit: this is duplicated for enum values, maybe just say once in the docblock of enum?